src/commands/environments/create.ts

```diff
@@ -38,19 +38,22 @@ export default class CreateEnvironment extends CreateCommand<CreateEnvironmentPa
 
         const { key, name, type, description, headless } = flags
 
-        await this.requireProject()
         if (headless && (!type || !key || !name)) {
             this.writer.showError('In headless mode, the type, key, and name flags are required')
             return
         }
 
-        const params = await this.populateParameters(CreateEnvironmentParams, false, {
-            key,
-            name,
-            description,
-            type,
-            headless
-        })
+        const params = await this.populateParameters(
+            CreateEnvironmentParams,
+            this.prompts,
+            {
+                key,
+                name,
+                description,
+                type,
+                headless
+            },
```

Comment:
any reason to not just pass `flags` here?

Reply:
that's how it was before but i don't see a reason why we can't just pass in `flags`. Will make the change

Reply:
figured out why it was being done this way. [mergeFlagsAndAnswers](https://github.com/DevCycleHQ/cli/blob/main/src/commands/createCommand.ts#L69) was iterating over the `flags` argument's keys to set a value for a key that didn't get set from the users flags using the prompt answers. Passing in an object with those keys explicitly meant that even if `key`, `name`, `description`, `type` were `undefined`, they would still exist in the object so the merge code was able to set the missing values correctly. When I changed it just passing `flags` directly, the provided answers weren't set on the final result because those keys I mentioned above didn't exist on the object. 

Anyways, as a fix, I've updated `mergeFlagsAndAnswers` to [iterate](https://github.com/DevCycleHQ/cli/pull/163/files#diff-d62b22b357d8189368367ee647935585998107d28bfaf657bac900e1087a9597R274) over the keys of `answers`, this way whatever doesn't get set by the flags, _will_ get set by the answers from the prompts.